From 8d99f27dbb2e1f7d3826cf9dfa45bdc4d8dc554e Mon Sep 17 00:00:00 2001
From: Nathaniel Shead <nathanieloshead@gmail.com>
Date: Thu, 25 Sep 2025 21:05:40 +1000
Subject: [PATCH] c++/modules: Remove incorrect assertion [PR122015,PR122019]

This assertion, despite what I said in r16-4070, is not valid: we can
reach here when deduping a VAR_DECL that didn't get a LANG_SPECIFIC in
the current TU.  It's still correct to always use lang_cplusplus however
as for anything else the decl would have been created with an
appropriate LANG_SPECIFIC to start with.

	PR c++/122015
	PR c++/122019

gcc/cp/ChangeLog:

	* module.cc (trees_in::install_entity): Remove incorrect
	assertion.

Signed-off-by: Nathaniel Shead <nathanieloshead@gmail.com>
(cherry picked from commit 4f9f1269f4e929ee471b717b47c33d2d5a3db0d3)
---
 gcc/cp/module.cc | 1 -
 1 file changed, 1 deletion(-)

diff --git a/gcc/cp/module.cc b/gcc/cp/module.cc
index 9ff2863d04eb..2f19b2b485dd 100644
--- a/gcc/cp/module.cc
+++ b/gcc/cp/module.cc
@@ -8114,7 +8114,6 @@ trees_in::install_entity (tree decl)
       if (!DECL_LANG_SPECIFIC (not_tmpl))
 	{
 	  maybe_add_lang_decl_raw (not_tmpl, false);
-	  gcc_checking_assert (!VAR_OR_FUNCTION_DECL_P (not_tmpl));
 	  SET_DECL_LANGUAGE (not_tmpl, lang_cplusplus);
 	}
       DECL_MODULE_ENTITY_P (not_tmpl) = true;
-- 
2.43.7


